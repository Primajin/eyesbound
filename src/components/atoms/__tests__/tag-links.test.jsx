import React from 'react';
import {create} from 'react-test-renderer';

import TagLinks from '../tag-links.jsx';

describe('TagLinks', () => {
	it('renders correctly with empty list', () => {
		const tags = [];
		const component = create(<TagLinks tags={tags}/>).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with tag list of one', () => {
		const tags = [
			{
				tag: {
					document: {
						data: {
							title: 'Hello',
						},
						uid: '1234-5678-90',
					},
				},
			},
		];
		const component = create(<TagLinks tags={tags}/>).toJSON();
		expect(component).toMatchSnapshot();
	});

	it('renders correctly with tag list of more', () => {
		const tags = [
			{
				tag: {
					document: {
						data: {
							title: 'Hello',
						},
						uid: '1234-5678-90',
					},
				},
			},
			{
				tag: {
					document: {
						data: {
							title: 'There',
						},
						uid: '9876-5432-10',
					},
				},
			},
		];
		const component = create(<TagLinks tags={tags}/>).toJSON();
		expect(component).toMatchSnapshot();
	});
});
