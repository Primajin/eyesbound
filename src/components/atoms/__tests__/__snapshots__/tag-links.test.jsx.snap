// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagLinks renders correctly with empty list 1`] = `null`;

exports[`TagLinks renders correctly with tag list of more 1`] = `
[
  <a
    href="/tag/1234-5678-90"
  >
    Hello
  </a>,
  " | ",
  <a
    href="/tag/9876-5432-10"
  >
    There
  </a>,
]
`;

exports[`TagLinks renders correctly with tag list of one 1`] = `
<a
  href="/tag/1234-5678-90"
>
  Hello
</a>
`;
